File: app/src/main/java/com/example/mrakopediareader/api/dto/Page.java
package com.example.mrakopediareader.api.dto;

public class Page {
    private final String title;

    private final String url;

    public Page(String title, String url) {
        this.title = title;
        this.url = url;
    }

    public String getTitle() {
        return title;
    }

    public String getUrl() {
        return url;
    }
}

File: app/src/main/java/com/example/mrakopediareader/api/dto/Page.kt
package com.example.mrakopediareader.api.dto

data class Page(val title: String, val url: String) {
    constructor() : this("", "")

    fun serialize(): String {
        return "$title$JOINER$url"
    }

    companion object {
        private const val JOINER = "|"

        fun parse(source: String): Page {
            val (title, url) = source.split(JOINER)
            return Page(title, url)
        }
    }
}